build.gradle.kts

```diff
@@ -0,0 +1,169 @@
+    maven("https://repo.papermc.io/repository/maven-public/")
+    maven("https://jitpack.io")
+    maven("https://maven.enginehub.org/repo/")
+    maven("https://repo.extendedclip.com/content/repositories/placeholderapi")
+    maven("https://nexus.neetgames.com/repository/maven-public")
+    maven("https://repo.walshy.dev/public")
+    maven("https://repo.codemc.io/repository/maven-public/")
+    maven("https://libraries.minecraft.net")
+}
+
+dependencies {
+    // Api is similar to "implementation" but that these deps are also exposed transitively
+    api("com.github.baked-libs.dough:dough-api:baf2d79f62")
+    api("io.papermc:paperlib:1.0.8")
+    api("commons-lang:commons-lang:2.6")
+
+    compileOnly("org.spigotmc:spigot-api:$minecraftVersion-R0.1-SNAPSHOT")
+    compileOnly("com.mojang:authlib:6.0.52")
+    compileOnly("com.google.code.findbugs:jsr305:3.0.2")
+
+    // Plugin dependencies
+    compileOnly("com.sk89q.worldedit:worldedit-core:7.2.19")
+    compileOnly("com.sk89q.worldedit:worldedit-bukkit:7.2.19")
+    compileOnly("com.gmail.nossr50.mcMMO:mcMMO:2.1.230")
+    compileOnly("me.clip:placeholderapi:2.11.5")
+    compileOnly("me.minebuilders:clearlag-core:3.1.6")
+    compileOnly("com.github.LoneDev6:itemsadder-api:3.6.1")
+    compileOnly("net.imprex:orebfuscator-api:5.4.0")
+
+
+    testRuntimeOnly("org.junit.platform:junit-platform-launcher")
+    testImplementation("org.junit.jupiter:junit-jupiter:5.10.2")
+    testImplementation("org.mockito:mockito-core:5.10.0")
+    testImplementation("org.slf4j:slf4j-simple:2.0.9")
+    testImplementation("com.github.seeseemelk:MockBukkit-v1.20:3.65.0")
+
+    // Override MockBukkit's Paper to a pinned slightly older version
+    // This is because MockBukkit currently fails after this PR: https://github.com/PaperMC/Paper/pull/9629
+    testImplementation("io.papermc.paper:paper-api:1.20.4-R0.1-20240205.114523-90")
+}
+
+group = "com.github.slimefun"
+version = property("project.version").toString()
+description = "Slimefun"
+
+java.toolchain.languageVersion.set(JavaLanguageVersion.of(targetJavaVersion))
+
+java {
+    withSourcesJar()
+}
+
+tasks {
+    withType<JavaCompile>().configureEach {
+        options.release = targetJavaVersion
+        options.encoding = encoding
+
+        // package info files are only important for Javadocs
+        // We can safely exclude them from the final jar -->
+        exclude("**/package-info.java")
+    }
+    withType<Javadoc>().configureEach {
+        options.encoding = encoding
+    }
+    javadoc {
+        setDestinationDir(file("javadocs"))
+        options {
+            title = "Slimefun4 - Javadocs"
+            windowTitle = "Slimefun4 - Javadocs"
+            // doclet = "org.gradle.external.javadoc.StandardJavadocDocletOptions"
+            // FIXME offline links, groups, and additional options.
+        }
+
+    }
+    assemble {
+        dependsOn(shadowJar)
+
+    }
```

Comment:
```suggestion
```
This bit is unneeded, `shadowJar` automatically assembles

Reply:
I'm pretty sure it doesnt. Without this calling `gradle build` will only build the unshaded jar. Nonetheless, we can remove this and just do `gradle shadowJar`

Reply:
Yep that's what I do with my gradle spigot plugins, no need to attach it to `build`